### PR TITLE
Explorer x500names

### PR DESCRIPTION
**Pull request by kasiastreich**
For demo purpose we still have short common name everywhere. We can change that behaviour later by specifying different formatting.

### CODE REVIEW

File: tools/explorer/src/main/kotlin/net/corda/explorer/views/Network.kt [hunk 1]
```diff
@@ -98,10 +99,11 @@ class Network : CordaView() {
 
         val button = button {
             graphic = vbox {
-                label(node.legalIdentity.name) { font = Font.font(font.family, FontWeight.BOLD, 15.0) }
+                label(PartyNameFormatter.short.format(node.legalIdentity.name)) { font = Font.font(font.family, FontWeight.BOLD, 15.0) }
                 gridpane {
                     hgap = 5.0
                     vgap = 5.0
+                    row("Full X500 name :") { label(PartyNameFormatter.full.format(node.legalIdentity.name)) }
```
Comment (rnicoll) [inline tools/explorer/src/main/kotlin/net/corda/explorer/views/Network.kt]: Is this targeted at a technical or non-technical audience? I'd be inclined to call it simply "Full name", I think very few people will understand what X500 is.

File: tools/explorer/src/main/kotlin/net/corda/explorer/views/TransactionViewer.kt [hunk 2]
```diff
@@ -132,8 +133,26 @@ class TransactionViewer : CordaView("Transactions") {
...
             column("Output", Transaction::outputs).cellFormat { text = it.toText() }
-            column("Input Party", Transaction::inputParties).cellFormat { text = it.flatten().map { it.value?.legalIdentity?.name }.filterNotNull().toSet().joinToString() }
-            column("Output Party", Transaction::outputParties).cellFormat { text = it.flatten().map { it.value?.legalIdentity?.name }.filterNotNull().toSet().joinToString() }
+            column("Input Party", Transaction::inputParties).setCustomCellFactory {
+                label {
+                    text = it.flatten().map {
+                        it.value?.legalIdentity?.let { PartyNameFormatter.short.format(it.name) }
+                    }.filterNotNull().toSet().joinToString()
+                    tooltip {
+                        text = it.flatten().map { it.value?.legalIdentity?.name }.filterNotNull().toSet().joinToString("\n")
```
Comment (rnicoll) [inline tools/explorer/src/main/kotlin/net/corda/explorer/views/TransactionViewer.kt]: Would it be worth having an extension method for `flatten().map { it.value?.legalIdentity?.name }.filterNotNull().toSet()`? For the text rendering, would mean formatting after reducing to a set, but might make this a bit easier to read.

File: tools/explorer/src/main/kotlin/net/corda/explorer/formatters/PartyNameFormatter.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import net.corda.core.crypto.Party
+import net.corda.core.crypto.commonName
+import org.bouncycastle.asn1.x500.X500Name
+
+object PartyNameFormatter {
+    val short = object : Formatter<String> {
+        override fun format(value: String) = X500Name(value).commonName
+    }
+
+    val full = object : Formatter<String> {
```
Comment (patrickkuo) [inline tools/explorer/src/main/kotlin/net/corda/explorer/formatters/PartyNameFormatter.kt]: This formatter is not doing anything, is this needed?

Comment (kasiastreich) [inline tools/explorer/src/main/kotlin/net/corda/explorer/formatters/PartyNameFormatter.kt]: I thought that we might want after the x500 changes specify how we show full names (in tooltips and for example Network view), so for now it's not doing much, but it's easier to substitute the formatting later.

**Pull request merged**